plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -191,7 +191,7 @@ private static Set<Location> listDirectories(TrinoFileSystem fileSystem, Locatio
     private static String listedDirectoryName(Location directory, Location location)
     {
         String prefix = directory.path();
-        if (!prefix.endsWith("/")) {
```

Comment:
`Fix bug in sync_partition_metadata() to handle table located in the root path of s3 bucket` -> `Add handling for root bucket table location in sync_partition_metadata`


Commit title 80 characters. Aim for conciseness.
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

Reply:
Thanks! I am done!

Reply:
Squash the commits into one pls and use a concise commit title.
You can take inspiration from my previous message.

Reply:
I squashed commits with a message, same as the title of this PR, following your guide! Thanks!

Reply:
I squashed commits with a message, same as the title of this PR, following your guide! Thanks!